crates/hir_ty/src/method_resolution.rs

```diff
@@ -385,7 +388,24 @@ fn iterate_method_candidates_impl(
                     krate,
                     traits_in_scope,
                     name,
-                    callback,
+                    &mut |ty, item, is_inherent| {
+                        if let (
+                            true,
+                            Some(from_module),
+                            Ty::Adt(AdtId(adt_id), _),
+                            AssocItemId::FunctionId(f),
+                        ) = (is_inherent, from_module, ty, item)
+                        {
+                            let adt_module = adt_id.module(def_db);
+                            let adt_resolver = adt_module.resolver(def_db);
+                            let fn_visibility =
+                                db.function_data(f).visibility.resolve(def_db, &adt_resolver);
+                            if !fn_visibility.is_visible_from(def_db, from_module) {
```

Comment:
Might use `mark::hit!();` here: 

https://github.com/rust-analyzer/rust-analyzer/blob/d54e1157b66017e4aae38328cd213286e39ca130/crates/test_utils/src/mark.rs

Reply:
Is this to make sure that this code path is actually executed? (Or aren't the tests enough?)

Reply:
Yes, to check that a test is covering this.